### PR TITLE
 pings peers before sending push messages (bp #28537 )

### PR DESCRIPTION
**Pull request by t-nelson**
v1.14 backport of #28537 

cc/ @behzadnouri

### CODE REVIEW

File: gossip/src/crds_gossip_push.rs [hunk 1]
```diff
@@ -353,17 +364,35 @@ impl CrdsGossipPush {
...
+            let now = Instant::now();
+            peers
+                .into_iter()
+                .filter_map(|(weight, peer)| {
+                    let node = (peer.id, peer.gossip);
+                    let (check, ping) = ping_cache.check(now, node, &mut pingf);
+                    if let Some(ping) = ping {
+                        pings.push((peer.gossip, ping));
+                    }
+                    if check {
```
Comment (t-nelson) [inline gossip/src/crds_gossip_push.rs:389]: only change for master is that `bool::then_some` is unstable in Rust 1.60

**Review by behzadnouri — APPROVED** [review]
lgtm

File: gossip/src/crds_gossip_push.rs (hunk 1 above)
Comment (t-nelson) [reply gossip/src/crds_gossip_push.rs]: switched to `bool::then` which is available

**Comment by t-nelson** [comment]
rebased. `automerge`ing

**Pull request merged**